### DISCUSSION TITLE
API docs?

### DISCUSSION THREAD

**Discussion by chriscamacho** [encode/starlette#1571] [Q&A] [answered]
Look as I might I could see no links to API docs, the descriptions and few simple examples barely gives me enough to start to evaluate starlette, certainly not enough to actually start using it ...

Take just one aspect of starlette, sessions - I could find one paragraph about it, but no API doc or even a simple usage example, even looking at the code - who knows what other values than "lax" could be used....?

**Comment by mjog** [comment]
Far more useful than a tutorial/guide with a few examples.

**Comment by mjog** [comment]
Oh, and [mkdocstrings](https://mkdocstrings.github.io/) makes automated API documentation generation for Python substantially less painful. It's not perfect, but you can at least integrate it into existing mkdoc documentation like Starlette's in a pretty straight-forward way.

**Comment by Kludex** [comment, marked as answer]
I've raised this discussion on Gitter. If it comes to something, I'll create an issue.
